### PR TITLE
OCPQE-21364: add/remove roles for cluster IAM service accounts in GCP host project

### PR DESCRIPTION
**Pull request by jianli-wei**
1. [OCPQE-21364](https://issues.redhat.com/browse/OCPQE-21364) Make sure proper service accounts additional permissions in place in Prow CI for GCP XPN

2. Insert the step `gcp-provision-iam-creds-xpn-roles` into UPI XPN provision chains, including
  - `cucushift-installer-rehearse-gcp-upi-private-xpn-provision`
  - `cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision`
  - `cucushift-installer-rehearse-gcp-upi-private-xpn-ingress-glb-provision`
  - `cucushift-installer-rehearse-gcp-upi-xpn-provision`

3. Insert the step `gcp-deprovision-iam-creds-xpn-roles` into UPI XPN deprovision chains, including
  - `cucushift-installer-rehearse-gcp-upi-private-xpn-deprovision`

4. Create the dedicated deprovision chain `cucushift-installer-rehearse-gcp-upi-xpn-deprovision` which has the step `gcp-deprovision-iam-creds-xpn-roles`

5. In the rehearsal test, [rehearse-59501-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-upi-private-xpn-sdn-f60-destructive #1864231096187621376](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59501/rehearse-59501-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-upi-private-xpn-sdn-f60-destructive/1864231096187621376), the cluster nodes do have the annotation `cloud.network.openshift.io/egress-ipconfig` along with non-empty value, see [the nodes.json](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/59501/rehearse-59501-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-upi-private-xpn-sdn-f60-destructive/1864231096187621376/artifacts/gcp-upi-private-xpn-sdn-f60-destructive/gather-extra/artifacts/nodes.json).

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-upi-private-xpn-sdn-f60-destructive

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-upi-private-xpn-sdn-f60-destructive

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-debug

**Comment by jianli-wei** [comment]
@jianlinliu @gpei please help review, thanks!

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/provision/cucushift-installer-rehearse-gcp-upi-xpn-provision-chain.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ chain:
   - chain: ipi-conf-gcp
   - ref: upi-conf-gcp
   - ref: upi-install-gcp
+  - ref: gcp-provision-iam-creds-xpn-roles
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/provision/cucushift-installer-rehearse-gcp-upi-xpn-provision-chain.yaml:9]: We also need to update `cucushift-installer-rehearse-gcp-upi-deprovision` accordingly, right?

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/provision/cucushift-installer-rehearse-gcp-upi-xpn-provision-chain.yaml:9]: No. Only for GCP shared VPC installations. BTW, our IPI XPN tests already have the steps.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/provision/cucushift-installer-rehearse-gcp-upi-xpn-provision-chain.yaml:9]: Oh, I meant `cucushift-installer-rehearse-gcp-upi-xpn-provision` is referenced in `cucushift-installer-rehearse-gcp-upi-xpn` workflow, here `gcp-provision-iam-creds-xpn-roles` is added into the provision chain, accordingly the deprovision chain for `cucushift-installer-rehearse-gcp-upi-xpn` should be updated accordingly to reference `gcp-deprovision-iam-creds-xpn-roles`, right?

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/provision/cucushift-installer-rehearse-gcp-upi-xpn-provision-chain.yaml:9]: Got it, updated the workflow `cucushift-installer-rehearse-gcp-upi-xpn`, thanks!

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/cucushift-installer-rehearse-gcp-upi-xpn-workflow.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ workflow:
     pre:
     - chain: cucushift-installer-rehearse-gcp-upi-xpn-provision
     post:
+    - ref: gcp-deprovision-iam-creds-xpn-roles
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/cucushift-installer-rehearse-gcp-upi-xpn-workflow.yaml:7]: How about introducing a dedicated deprovision chain for this workflow.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/deprovision/cucushift-installer-rehearse-gcp-upi-xpn-deprovision-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+chain:
+  as: cucushift-installer-rehearse-gcp-upi-xpn-deprovision
+  steps:
+  - ref: gcp-deprovision-iam-creds-xpn-roles
+  - chain: ipi-deprovision
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/xpn/deprovision/cucushift-installer-rehearse-gcp-upi-xpn-deprovision-chain.yaml:5]:
```suggestion
  - chain: cucushift-installer-rehearse-gcp-upi-deprovision
```

**Pull request merged**